new-components/targets/git-clone/pkg/git/git.go

```diff
@@ -0,0 +1,207 @@
+package git
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/go-errors/errors"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/transport/http"
+	"github.com/smithy-security/pkg/env"
+)
+
+const errInvalidConfigurationReasonEmpty = "cannot be empty"
+
+type (
+	// Conf wraps the component configuration.
+	Conf struct {
+		ClonePath  string
+		RepoURL    string
+		BranchName string
```

Comment:
this is missing a ref, i understand this is an MVP let's just add a ticket to implement this whenever someone asks for it

Reply:
Changed to ref.